src/js/components/DataTable/doc.js

```diff
@@ -163,6 +169,13 @@ export const doc = DataTable => {
       browser, such as columns.search, sortable, groupBy, or 
       columns.aggregate.`,
     ),
+    pin: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.oneOf(['header', 'footer']),
+    ]).description(
+      `Whether the header and/or footer should be pinned when
+      not all rows are visible.`,
+    ),
```

Comment:
It is not clear what is the default behavior when `pin=true`.
Also, how do you set both to be pinned if the proptypes are limiting to a string or a bool?

Reply:
added more detail